src/snowflake/connector/result_batch.py

```diff
@@ -568,11 +564,9 @@ def _load(
         """
         from .arrow_iterator import PyArrowIterator
 
-        gfd = GzipFile(fileobj=response.raw, mode="r")
-
         iter = PyArrowIterator(
             None,
-            gfd,
+            io.BytesIO(response.content),
```

Comment:
Our C-extension requires a file like object, so we wrap it in `io.BytesIO`. Note that `.content` is automatically decompressed too